snapcraft/tests/pluginhandler/test_pluginhandler.py

```diff
@@ -1821,6 +1821,54 @@ def test_clean_part_remaining_parts(self, mock_exists, mock_listdir,
         mock_listdir.assert_called_once_with(partdir)
         self.assertFalse(mock_rmdir.called)
 
+    def test_clean_dir_symlinks(self):
+        # Create part1 and get it through the "build" step.
+        handler1 = mocks.loadplugin('part1')
+        handler1.makedirs()
+
+        bindir = os.path.join(handler1.code.installdir, 'bin')
+        os.makedirs(bindir)
+        path = os.path.join(bindir, '1')
+        os.mkdir(path)
+        symlink_path = os.path.join(bindir, '1-link')
+        os.symlink(path, symlink_path)
```

Comment:
Note that this is a broken symlink. You probably want `os.symlink('1', symlink_path)`.